### PR TITLE
feat (angular) : Jest test setup  in angular package library.

### PR DESCRIPTION
**Pull request by SantoshLatpate**
#### Description of changes
1. Remove jasmine and karma from the present angular library package.
2. Setup jest in angular library package.
3. Write basic 2 test suit in angular library.

#### Checklist

- [ ] PR description included
- [ ] `yarn test` passes
- [ ] Tests are updated
- [ ] No side effects or [`sideEffects`](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/CONTRIBUTING.md#code-standards) field updated
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/angular/projects/ui-angular/src/lib/primitives/checkbox/checkbox.component.spec.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+describe('CheckboxComponent', () => {
+  let fixture: CheckboxComponent;
+
+  beforeEach(() => {
+    fixture = new CheckboxComponent();
+  });
+  it('Default checkbox behaviour -> ', () => {
+    expect(fixture.isChecked).toEqual(false);
+  });
+});
```
Comment (ErikCH) [inline packages/angular/projects/ui-angular/src/lib/primitives/checkbox/checkbox.component.spec.ts:12]: This is a great first start to adding in testing!

Could you also add in the testing library? https://testing-library.com/docs/angular-testing-library/intro

And convert these tests to use that instead? That way we have consistency between all 3 platforms.

Comment (SantoshLatpate) [reply packages/angular/projects/ui-angular/src/lib/primitives/checkbox/checkbox.component.spec.ts:12]: Ok, I will look into this.

**Comment by wlee221** [comment]
This PR should target `main`, and not `poc/angular-primitives/alpha`. Can you readjust? 🙏

File: packages/angular/package.json [hunk 2]
```diff
@@ -7,38 +7,33 @@
...
-    "@angular/compiler-cli": "~11.2.4",
     "@angular/compiler": "~11.2.4",
+    "@angular/compiler-cli": "~11.2.4",
     "@angular/core": "~11.2.4",
-    "@angular/platform-browser-dynamic": "~11.2.4",
     "@angular/platform-browser": "~11.2.4",
+    "@angular/platform-browser-dynamic": "~11.2.4",
     "@angular/router": "~11.2.4",
-    "@types/jasmine": "~3.6.0",
+    "@types/jest": "^28.1.6",
```
Comment (wlee221) [inline packages/angular/package.json:30]: ✨

File: packages/angular/projects/ui-angular/src/lib/primitives/checkbox/checkbox.component.spec.ts (hunk 1 above)
Comment (wlee221) [reply packages/angular/projects/ui-angular/src/lib/primitives/checkbox/checkbox.component.spec.ts:12]: Same comment as @ErikCH

File: packages/angular/projects/ui-angular/package.json [hunk 3]
```diff
@@ -18,15 +18,29 @@
     "aws-amplify": "3.x.x || 4.x.x"
   },
   "dependencies": {
-    "@aws-amplify/ui-components": "^1.7.0",
+    "@angular-builders/jest": "^14.0.0",
```
Comment (wlee221) [inline packages/angular/projects/ui-angular/package.json:21]: This should be a `devDependency` at `packages/angular/package.json`.

File: packages/angular/projects/ui-angular/package.json [hunk 4]
```diff
@@ -18,15 +18,29 @@
...
     "@aws-amplify/ui": "3.12.4",
+    "@aws-amplify/ui-components": "^1.7.0",
```
Comment (wlee221) [inline packages/angular/projects/ui-angular/package.json:23]: Why is this added?

File: packages/angular/projects/ui-angular/package.json [hunk 5]
```diff
@@ -18,15 +18,29 @@
...
     "@stencil/core": "2.8.0",
+    "@testing-library/angular": "^12.1.0",
```
Comment (wlee221) [inline packages/angular/projects/ui-angular/package.json:25]: This should be a devDependency at packages/angular/package.json.

File: packages/angular/projects/ui-angular/package.json [hunk 6]
```diff
@@ -18,15 +18,29 @@
...
+    "preset": "jest-preset-angular",
+    "setupFilesAfterEnv": [
+      "<rootDir>/src/setup.jest.ts"
+    ]
+  },
+  "devDependencies": {
+    "@types/jest": "^28.1.6",
+    "jest": "^28.1.3",
+    "jest-environment-jsdom": "^28.1.3",
+    "jest-preset-angular": "^12.2.0"
```
Comment (wlee221) [inline packages/angular/projects/ui-angular/package.json:44]: Same as above -- this should be a devDependency at packages/angular/package.json if possible.

File: packages/angular/projects/ui-angular/package.json (hunk 4 above)
Comment (wlee221) [reply packages/angular/projects/ui-angular/package.json:23]: Oops nvm, this is just a sorting change! my bad

File: packages/angular/projects/ui-angular/package.json [hunk 7]
```diff
@@ -18,15 +18,29 @@
...
   },
   "publishConfig": {
     "directory": "../../dist/ui-angular"
+  },
+  "jest": {
+    "preset": "jest-preset-angular",
+    "setupFilesAfterEnv": [
+      "<rootDir>/src/setup.jest.ts"
+    ]
+  },
```
Comment (wlee221) [inline packages/angular/projects/ui-angular/package.json:39]: Wonder if this can be moved out to parent dir as well.

**Comment by wlee221** [comment]
Closing, Angular has to be bumped to 12 first

**Pull request closed without merging**